syncd/usr/bin/syncd_init_common.sh

```diff
@@ -0,0 +1,116 @@
+fi
+
+case "$(cat /proc/cmdline)" in
+  *fast-reboot*)
+     FAST_REBOOT='yes'
+    ;;
+  *)
+     FAST_REBOOT='no'
+    ;;
+esac
+
+
+config_syncd_bcm()
+{
+    CMD_ARGS+=" -p $HWSKU_DIR/sai.profile"
+
+    [ -e /dev/linux-bcm-knet ] || mknod /dev/linux-bcm-knet c 122 0
+    [ -e /dev/linux-user-bde ] || mknod /dev/linux-user-bde c 126 0
+    [ -e /dev/linux-kernel-bde ] || mknod /dev/linux-kernel-bde c 127 0
+
+    if [ $FAST_REBOOT == "yes" ]; then
+        CMD_ARGS+=" -t fast"
+    fi
+}
+
+config_syncd_mlnx()
+{
+    CMD_ARGS+=" -p /tmp/sai.profile"
+
+    [ -e /dev/sxdevs/sxcdev ] || ( mkdir -p /dev/sxdevs && mknod /dev/sxdevs/sxcdev c 231 193 )
+
+    # Read MAC address and align the last 6 bits.
+    MAC_ADDRESS=`ip link show eth0 | grep ether | awk '{print $2}'`
+    last_byte=`python -c "print '$MAC_ADDRESS'[-2:]"`
+    aligned_last_byte=`python -c "print format(int(int('$last_byte', 16) & 0b11000000), '02x')"`  # put mask and take away the 0x prefix
+    ALIGNED_MAC_ADDRESS=`python -c "print '$MAC_ADDRESS'[:-2] + '$aligned_last_byte'"`          # put aligned byte into the end of MAC
+
+    # Write MAC address into /tmp/profile file.
+    cat $HWSKU_DIR/sai.profile > /tmp/sai.profile
+    echo "DEVICE_MAC_ADDRESS=$ALIGNED_MAC_ADDRESS" >> /tmp/sai.profile
+}
+
+config_syncd_centec()
+{
+    CMD_ARGS+=" -p /tmp/sai.profile"
+
+    [ -e /dev/linux_dal ] || mknod /dev/linux_dal c 198 0
+    [ -e /dev/net/tun ] || ( mkdir -p /dev/net && mknod /dev/net/tun c 10 200 )
+
+    # Read MAC address and align the last 6 bits.
+    MAC_ADDRESS=`ip link show eth0 | grep ether | awk '{print $2}'`
+    last_byte=`python -c "print '$MAC_ADDRESS'[-2:]"`
+    aligned_last_byte=`python -c "print format(int(int('$last_byte', 16) & 0b11000000), '02x')"`  # put mask and take away the 0x prefix
+    ALIGNED_MAC_ADDRESS=`python -c "print '$MAC_ADDRESS'[:-2] + '$aligned_last_byte'"`          # put aligned byte into the end of MAC
+
+    # Write MAC address into /tmp/profile file.
+    cat $HWSKU_DIR/sai.profile > /tmp/sai.profile
+    echo "DEVICE_MAC_ADDRESS=$ALIGNED_MAC_ADDRESS" >> /tmp/sai.profile
+}
+
+config_syncd_cavium()
+{
+    CMD_ARGS+=" -p /etc/ssw/AS7512/profile.ini -N"
+
+    export XP_ROOT=/usr/bin/
+
+    # Wait until redis-server starts
+    until [ $(redis-cli ping | grep -c PONG) -gt 0 ]; do
+        sleep 1
+    done
+
+    redis-cli FLUSHALL
+}
+
+config_syncd()
+{
+    if [ $SONIC_ASIC_TYPE == "broadcom" ]; then
+        config_syncd_bcm
+    elif [ $SONIC_ASIC_TYPE == "mellanox" ]; then
+        config_syncd_mlnx
+    elif [ $SONIC_ASIC_TYPE == "cavium" ]; then
+        config_syncd_cavium
+    elif [ $SONIC_ASIC_TYPE == "centec" ]; then
+        config_syncd_centec
+    else
+        echo "Unknown ASIC type $SONIC_ASIC_TYPE"
+        exit 1
+    fi
+
+    if [ ${ENABLE_SAITHRIFT} == 1 ]; then
+        CMD_ARGS+=" -r -m $HWSKU_DIR/port_config.ini"
+    fi
+
+    [ -r $PLATFORM_DIR/syncd.conf ] && . $PLATFORM_DIR/syncd.conf
+}
+
```

Comment:
http://stackoverflow.com/questions/8748831/when-do-we-need-curly-braces-in-variables-using-bash